Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -89,6 +89,26 @@ namespace AZ::Dom
         return !operator==(key);
     }
 
+    bool PathEntry::operator<(const PathEntry& rhs) const
+    {
+        return AZStd::visit(
+            [&](auto&& lhsValue)
+            {
+                using CurrentType = AZStd::decay_t<decltype(lhsValue)>;
+                if constexpr (AZStd::is_same_v<CurrentType, size_t>)
+                {
+                    const size_t* rhsValue = AZStd::get_if<size_t>(&rhs.m_value);
+                    return rhsValue == nullptr ? true : lhsValue < *rhsValue;
```

Comment:
Why return true if the types don't match?

Reply:
Switched this out for a hash, but it was just to avoid treating disparate types as equal.